src/components/Select.tsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import classNames from 'classnames';
+import React, { useEffect, useState, useRef } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { PerPage } from '../types/PerPage';
+import { SortBy } from '../types/SortBy';
+import { getSearchWith } from '../utils/getSearchWith';
+
+type Props = {
+  options: PerPage [] | SortBy[],
+  current: PerPage | SortBy | string,
+  title: string;
+  searchPar: string;
+  isLong: boolean;
+};
+
+export const Select: React.FC<Props> = ({
+  options,
+  current,
+  title,
+  searchPar,
+  isLong,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchParams] = useSearchParams();
+  const dropDownRef = useRef<HTMLDivElement | null>(null);
+  const optionsRef = useRef<HTMLUListElement | null>(null);
+  let timerId: NodeJS.Timeout;
+
+  const handleClickDropDown = () => {
+    timerId = setTimeout(() => {
+      setIsOpen(currentState => !currentState);
+    }, 200);
+  };
```

Comment:
use `useRef` for `timerId`
```suggestion
  const timerId = useRef(0);

  const handleClickDropDown = () => {
    timerId.current = setTimeout(() => {
      setIsOpen(currentState => !currentState);
    }, 200);
  };
```